blockly/src/systems/TintTilesSystem.java

```diff
@@ -0,0 +1,136 @@
+package systems;
+
+import components.TintDirectionComponent;
+import core.Entity;
+import core.Game;
+import core.System;
+import core.components.PositionComponent;
+import core.level.Tile;
+import core.level.elements.ILevel;
+import core.level.utils.Coordinate;
+import core.utils.components.MissingComponentException;
+import java.util.HashSet;
+import java.util.Set;
+import utils.Direction;
+import utils.components.ai.fight.StraightRangeAI;
+
+/**
+ * The TintTilesSystem is responsible for applying a tinting effect to tiles in the range of an
+ * entity.
+ *
+ * <p>Entities with the {@link TintDirectionComponent} will be processed by this system.
+ *
+ * @see TintDirectionComponent
+ * @see StraightRangeAI
+ */
+public class TintTilesSystem extends System {
+
+  /**
+   * Creates a new TintTilesSystem.
+   *
+   * <p>This system processes all entities with a {@link TintDirectionComponent} and applies the
+   * tinting effect to the tiles in the range of the entity.
+   *
+   * <p>It also removes the tinting effect when the entity is removed.
+   */
+  public TintTilesSystem() {
+    super(TintDirectionComponent.class);
```

Comment:
```suggestion
    super(TintDirectionComponent.class,PositionComponent.class);
```

dann verarbeitet er nur Entitäten die beides haben (sonst fliegt er dir unten hoch)